### PR TITLE
Add timer for limit between submissions and error message box

### PR DESCRIPTION
**Pull request by shsimeonova**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/343

### CODE REVIEW

**Review by PetarNeshkov — APPROVED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/submission-box/SubmissionBox.tsx [hunk 1]
```diff
@@ -1,28 +1,43 @@
...
+import Countdown, { ICountdownRemainingType, Metric } from '../../guidelines/countdown/Countdown';
 import Heading, { HeadingType } from '../../guidelines/headings/Heading';
 import List, { Orientation } from '../../guidelines/lists/List';
 import ExecutionTypeSelector from '../execution-type-selector/ExecutionTypeSelector';
 
 import styles from './SubmissionBox.module.scss';
 
 const SubmissionBox = () => {
-    const { actions: { selectSubmissionTypeById } } = useSubmissions();
+    const [ showSubmissionLimitTimer, setShowSubmissionLimitTimer ] = useState<boolean>(true);
```
Comment (Minkov) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/submission-box/SubmissionBox.tsx:21]: Isn't this the same as:
```
const showSubmissionLimitTimer = useMemo(
 () => submitLimit > 0,
 [submitLimit],
);
```

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/submission-box/SubmissionBox.tsx [hunk 2]
```diff
@@ -85,11 +100,95 @@ const SubmissionBox = () => {
...
+            setSubmitLimit(contest.limitBetweenSubmissions as number);
+        }
+    }, [ contest ]);
+
     const handleOnSubmit = useCallback(async () => {
         await submit();
         updateSubmissionCode('');
     }, [ submit, updateSubmissionCode ]);
 
+    const submissionLimitCountdownRender = useCallback((remainingTime: ICountdownRemainingType) => {
```
Comment (Minkov) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/submission-box/SubmissionBox.tsx:119]: `renderSubmissionLimitCountdown`

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/guidelines/alert-box/AlertBox.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    const [ hide, setHide ] = useState<boolean>(false);
+
+    const internalClassName = useMemo(() => {
+        const baseClsName = concatClassNames(alertBoxTypeToDefaulClassName[type], styles.alertBox, className);
+
+        if (hide) {
+            return concatClassNames(styles.hidden, baseClsName);
+        }
+
+        return baseClsName;
```
Comment (Minkov) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/guidelines/alert-box/AlertBox.tsx:40]:
```
return hide
    ? concatClassNames(styles.hidden, baseClsName)
    : baseClsName;
```

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/guidelines/alert-box/AlertBox.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    onClose: () => void;
+}
+
+const AlertBox = ({
+    message,
+    type,
+    className,
+    onClose,
+}: IAlertBoxProps) => {
+    const [ hide, setHide ] = useState<boolean>(false);
```
Comment (Minkov) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/guidelines/alert-box/AlertBox.tsx:31]: `isHidden`?

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx [hunk 5]
```diff
@@ -43,6 +47,8 @@ const SubmissionsProvider = ({ children }: ISubmissionsProviderProps) => {
     const [ selectedSubmissionType, setSelectedSubmissionType ] =
         useState<ISubmissionTypeType | null>(defaultState.state.selectedSubmissionType);
     const [ submissionCode, setSubmissionCode ] = useState<string>(defaultState.state.submissionCode);
+    const [ submitMessage, setSubmitMessage ] = useState<string | null>(null);
+    const [ submitSuccessful, setSubmitSuccessful ] = useState<boolean | null>(null);
```
Comment (Minkov) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx:51]: `const isSubmissionSucessful = useMemo(() => isSuccess, [isSuccess])`

File: Services/UI/OJS.Services.Ui.Business/Implementations/ParticipantsBusinessService.cs [hunk 6]
```diff
@@ -144,6 +147,9 @@ public async Task<ServiceResult<string>> UpdateParticipationEndTimeByIdAndTimeIn
             return ServiceResult<ICollection<string>>.Success(invalidForUpdateParticipantUsernames);
         }
 
+        public Task<int> GetParticipantLimitBetweenSubmissions(int participantId, int contestLimitBetweenSubmissions)
+            => Task.FromResult(this.submissionsData.GetUserSubmissionTimeLimit(participantId, contestLimitBetweenSubmissions));
```
Comment (Minkov) [inline Services/UI/OJS.Services.Ui.Business/Implementations/ParticipantsBusinessService.cs:151]:
```
this.submissionsData.GetUserSubmissionTimeLimit(participantId, contestLimitBetweenSubmissions)
 .ToTask()
```

**Review by Minkov — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/submission-box/SubmissionBox.tsx (hunk 1 above)
Comment (shsimeonova) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/submission-box/SubmissionBox.tsx:21]: yes

**Review by Minkov — APPROVED** [review]

**Review by PetarNeshkov — APPROVED** [review]

**Pull request merged**